src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -506,9 +510,10 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
     CORINFO_InstructionSet isa = HWIntrinsicInfo::lookupIsa(intrinsic);
 
-    if ((isa == InstructionSet_Vector256) && !compExactlyDependsOn(InstructionSet_AVX))
+    if (((isa == InstructionSet_Vector256) && !compExactlyDependsOn(InstructionSet_AVX)) ||
+        (((isa == InstructionSet_Vector512) && !IsBaselineVector512IsaSupported())))
```

Comment:
I don't think either of these checks are actually needed since we should've returned `NI_Illegal` in the intrinsic lookup.